### PR TITLE
Fix compile errors.

### PR DESCRIPTION
**Pull request by groutr**
This resolves the issues with missing symbols on my machine as well as
the troubles with the netcdf library.

The root of the issue was the hardcoding of gfortran in the makefiles. The netcdf library issues were resolved by querying netcdf library for the include paths and linking options.

tag: @hellkite500 @jameshalgren

### CODE REVIEW

**Review by jameshalgren — COMMENTED** [review]
What would it look like to add a simple test to check if $F90 is empty, either in the makefile or in the compiler.sh script?

**Comment by jameshalgren** [comment]
Tested on gfortran -- works. 

Compiles but produces the following error with intel compiler stack on Cheyenne (not a reason not to merge the PR.)

```
python compute_nhd_routing_SingleSeg_v02.py -f ../../test/input/yaml/CustomInput.yaml
Traceback (most recent call last):
  File "compute_nhd_routing_SingleSeg_v02.py", line 297, in <module>
    import fast_reach
ImportError: /glade/u/home/user/t-route/src/python_routing_v02/fast_reach.cpython-37m-x86_64-linux-gnu.so: undefined symbol: __intel_sse2_strlen
```

File: src/python_routing_v02/compiler.sh [hunk 1]
```diff
@@ -10,6 +10,12 @@ build_reservoir_kernel=true
 build_framework=true
 build_routing=true
 
+if [ -z "$F90" ]
+then
+    export F90="gfortran"
+    echo "using F90='gfortran'"
+fi
+
```
Comment (jameshalgren) [inline src/python_routing_v02/compiler.sh:18]: @groutr or @hellkite500 
Will you comment on the palatability of this insertion to catch the case where no F90 environment variable is in place?

Thanks...

Comment (hellkite500) [reply src/python_routing_v02/compiler.sh:18]: This is really just a dev utility, I don't see too much harm imposing a default.  Worst case, gfortran doesn't exist and a make error occurs.

**Review by jameshalgren — APPROVED** [review]

**Review by jameshalgren — APPROVED** [review]
Rebased. Approved.
